### DISCUSSION TITLE
Custom activity property always null

### DISCUSSION THREAD

**Discussion by james987654321** [elsa-workflows/elsa-core#1234] [Q&A] [answered]
Since moving to V2 I have been unable to get the value of the properties in my custom activity below.  I'm sure I have done something stupid but cant work out what, can you suggest anything???

[Activity(Category = "Custom Activities", Description = "Update an item", Outcomes = new[]{ OutcomeNames.Done, "Not Found" })]
    public class UpdateItem : Activity
    {
        public UpdateItem()        {        }

        [ActivityProperty(Hint = "Enter ID of the item to update.", SupportedSyntaxes = new[] { SyntaxNames.Literal, SyntaxNames.Liquid, SyntaxNames.JavaScript })]
        public string itemID { 
            get => GetState<string>();
            set => SetState(value);
         }

        [ActivityProperty(Hint = "Enter the property to update.", SupportedSyntaxes = new[] { SyntaxNames.Literal, SyntaxNames.Liquid, SyntaxNames.JavaScript })]
        public string itemProperty { 
            get => GetState<string>();
            set => SetState(value);
         }

        [ActivityProperty(Hint = "Enter the value to set.", SupportedSyntaxes = new[] { SyntaxNames.Literal, SyntaxNames.Liquid, SyntaxNames.JavaScript })]
        public string itemValue  { 
            get => GetState<string>();
            set => SetState(value);
         }
        
        [ActivityProperty(Hint = "Enter the table name.", , SupportedSyntaxes = new[] { SyntaxNames.Literal, SyntaxNames.Liquid, SyntaxNames.JavaScript })]
        public string tableName  { 
            get => GetState<string>();
            set => SetState(value);
         }
        
        protected override async ValueTask<IActivityExecutionResult> OnExecuteAsync(ActivityExecutionContext context)
        {                      
            
            switch (tableName)
            {
                case "NewUser": 
                    //Do someting.....
                    
                default:
                    break;
            }        
            
            return Outcome("Not Found");
        }
}

**Comment by sfmskywalker** [comment, marked as answer]
Perhaps there’s a bug preventing `SetState` and or `GetState` from functioning correctly.

As of Elsa 2, you can implement activity properties without using these methods.

Perhaps give that a go first. If that works, then it confirms the bug I mentioned.

**Reply by james987654321** [reply]
get the same with the below... 

[ActivityProperty(Hint = "Enter the table name")]
 public string testPropertyInput  { get; set; }

**Reply by sfmskywalker** [reply]
Can you send me a repro that I can look at? There must be something else that is missing not seen here.

**Reply by james987654321** [reply]
created a new basic project just to show the issue, published to GitHub before so hope this works :)

https://github.com/james987654321/ElsaCustomActivity.git

**Reply by sfmskywalker** [reply]
Excellent. I was able to reproduce the issue with your project.

As it turns out, the issue is quite an unexpected one: your properties are `camelCased` and not `PascalCased`.

Why does this matter, you ask? Great question 😄 

What happens is that Elsa uses coded activities as a source for the `TypeBasedActivityProvider` that yields activity type objects. An activity type contains all the data there is to know about an activity, including its properties.

Here's the issue: when storing the name of the property, it takes that name **and then Pascalizes** it as you can see [here](https://github.com/elsa-workflows/elsa-core/blob/master/src/core/Elsa.Core/Metadata/TypedActivityTypeDescriber.cs#L88).

If you rename your property to `TestPropertyInput` then it just works.

I will see if I can simply remove the Pascalization of property names, because even though you should follow standard naming convention and Pascal-case your C# properties, Elsa shouldn't break if you don't follow that naming convention.

**Reply by james987654321** [reply]
excellent thanks all working in my project now, told you it would be something stupid I had done :D one of them bad habits I've never got out of.
